crates/core_simd/src/vector.rs

```diff
@@ -220,18 +208,7 @@ where
 
     /// Converts a SIMD vector to an array.
     pub const fn to_array(self) -> [T; N] {
-        let mut tmp = core::mem::MaybeUninit::uninit();
-        // SAFETY: writing to `tmp` is safe and initializes it.
-        //
-        // FIXME: We currently use a pointer store instead of `transmute_copy` because `repr(simd)`
-        // results in padding for non-power-of-2 vectors (so vectors are larger than arrays).
-        //
-        // NOTE: This deliberately doesn't just use `self.0`, see the comment
-        // on the struct definition for details.
-        unsafe {
-            self.store(tmp.as_mut_ptr());
-            tmp.assume_init()
-        }
+        *self.as_array()
```

Comment:
this doesn't generate a vector `store` instruction so llvm may produce less optimal code.